src/smsCommandFields/FieldDataElementWithCategoryOptionComboAddFormulaButton.js

```diff
@@ -0,0 +1,82 @@
+import { Button, ReactFinalForm } from '@dhis2/ui'
+import { PropTypes } from '@dhis2/prop-types'
+import { useDataEngine } from '@dhis2/app-runtime'
+import React, { useEffect, useState } from 'react'
+import i18n from '../locales'
+import styles from './FieldDataElementWithCategoryOptionComboAddFormulaButton.module.css'
+
+const { useField } = ReactFinalForm
+
+const DATA_ELEMENTS_QUERY = {
+    dataElement: {
+        resource: 'dataElements',
+        params: ({ code }) => ({
+            filter: `code:eq:${code}`,
+            fields: 'displayName',
+            paging: 'false',
+        }),
+    },
+}
+
+export const FieldDataElementWithCategoryOptionComboAddFormulaButton = ({
+    baseName,
+    onClick,
+    disabled,
+}) => {
+    const engine = useDataEngine()
+    const [called, setCalled] = useState(false)
+    const [loading, setLoading] = useState(false)
+    const [formulaDataElementName, setFormulaDataElementName] = useState('')
+
+    const smsCode = useField(baseName, {
+        subscription: { value: true },
+    }).input.value
+
+    const { formula, code } = smsCode
+    const operator = formula && formula[0]
+    const dataElementCode = formula && formula.slice(1)
+
+    useEffect(() => {
+        if (!called && dataElementCode) {
+            setLoading(true)
+            setCalled(true)
+
+            engine
+                .query(DATA_ELEMENTS_QUERY, {
+                    variables: { code: dataElementCode },
+                })
+                .then(response => {
+                    const [{ displayName }] = response.dataElement.dataElements
+                    setFormulaDataElementName(displayName)
+                })
+                .finally(() => setLoading(false))
+        }
+    }, [dataElementCode, called])
+
+    return (
+        <>
+            {loading && i18n.t('Loading formula')}
```

Comment:
I think we have different stylistic preferences here, I've noticed this pattern throughout the app. Instead of a single return with conditional parts I prefer a return per (similar) condition. That way I feel the code is easier to parse, and it's less easy to make mistakes in handling conditions.

It can introduce jsx duplication, but that can be handled.